### ISSUE TITLE
igb module wont go past 100mb/s on a gigabit NIC

### ISSUE THREAD

**Issue by fiercebrake** [Frogging-Family/linux-tkg#317]
Greetings everyone!

I am runing archlinux + kernel 5.10

turns out my NIC will not go past 100Mb/s

I already made sure the cable is not the issue, and that the modem supports such speed. Everything is fine on those 2. 

THe same box running windows, has no problems going 1000mb/s, problem is on linux.

The NIC is an integrated 'Intel Corporation I211 Gigabit Network Connection' on an ASUS MOBO

anyone has ny clue/hint as to how go bout this particular problem?

based on some google results, i already tried forcing the NIC via CLI to run at 1k, but then the NIC fails to connect to the modem.

**Comment by Tk-Glitch** [comment]
So far ethernet negotiation has always defaulted to 100mbps on linux for me. Going through my network manager to set it to 1gbps full duplex works fine on all my machines though. I never tried doing it through CLI though.

![Screenshot_20210921_215835](https://user-images.githubusercontent.com/741977/134239183-15798e43-46e9-4acf-8996-7531aecda8e3.png)

**Comment by fiercebrake** [comment]
Yeah,

Manually setting the NIC to 1k either via CLI or GUI will cause the NIC to be unable to connect to the modem/router

**Comment by Tk-Glitch** [comment]
I guess you could try using the `igb-dkms` (AUR) package to see if it's of any help. That specific adapter seems to be quite problematic for many (independently of OS used), so it might have special needs.

**Comment by fiercebrake** [comment]
Yeah,

I tried that a few times before, and based on your comment i did again, but the install fails with the error below. Any advise on how could i fix that ?


"DKMS make.log for igb-5.3.5.3 for kernel 5.14.6-201-tkg-pds (x86_64)
Wed Sep 22 08:23:37 AM CST 2021
cc1: error: code model kernel does not support PIC mode
/bin/sh: line 1: [: -ge: unary operator expected
Makefile:211: *** *** Aborting the build. *** This driver '' is not supported on kernel versions older than 2.4.21, because this driver requires NAPI support..  Stop."

**Comment by timocapa** [comment]
> ethernet negotiation has always defaulted to 100mbps on linux for me

Do you actually get a 1 GBe link? Auto negotiation should work just fine - does for me, at least.

![image](https://user-images.githubusercontent.com/22844613/135330927-dc7a66fa-aa01-42fc-8ada-c338d45ef17a.png)

**Comment by Tk-Glitch** [comment]
Yes I do. I admit I haven't tried it with my current motherboard (b550), but all my older machines needed to be set to 1gbps manually to get that speed.
